jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java

```diff
@@ -0,0 +1,317 @@
+         * specified to the service provider.
+         *
+         * @param mimeType A String representation of the jcr:mimeType.
+         * @return The calling instance.
+         * @see <a href="https://docs.adobe.com/content/docs/en/spec/jcr/2.0/3_Repository_Model.html#3.7.11.10%20mix:mimeType">
+         *     JCR 2.0 Repository Model - jcr:mimeType</a>
+         */
+        @NotNull
+        public BinaryDownloadOptionsBuilder withMimeType(@NotNull String mimeType) {
+            this.mimeType = mimeType;
+            return this;
+        }
+
+        /**
+         * Sets the encoding of the {@link BinaryDownloadOptions} object to be
+         * built.  This value should be a valid {@code jcr:encoding}.
+         * <p>
+         * Calling this method has the effect of instructing the service
+         * provider to set {@code encoding} as the encoding of the content type
+         * in the {@code Content-Type} header of the response to a request
+         * issued with a URI obtained by calling {@link
+         * BinaryDownload#getURI(BinaryDownloadOptions)}.  This value can be
+         * later retrieved by calling {@link
+         * BinaryDownloadOptions#getEncoding()} on the instance returned by a
+         * call to {@link #build()}.
+         * <p>
+         * Note that setting the encoding only makes sense if the MIME type has
+         * also been set to a text-based MIME type.  See {@link
+         * #withMimeType(String)}.
+         * <p>
+         * The caller should ensure that the proper encoding has been set for
+         * the MIME type; the implementation does not perform any validation of
+         * these settings.
+         *
+         * @param encoding A String representation of the jcr:encoding.
+         * @return The calling instance.
+         * @see <a href="https://docs.adobe.com/content/docs/en/spec/jcr/2.0/3_Repository_Model.html#3.7.11.10%20mix:mimeType">
+         *     JCR 2.0 Repository Model - jcr:encoding</a>
+         */
+        @NotNull
+        public BinaryDownloadOptionsBuilder withEncoding(@NotNull String encoding) {
+            this.encoding = encoding;
+            return this;
+        }
+
+        /**
+         * Sets the filename of the {@link BinaryDownloadOptions} object to be
+         * built.
+         * <p>
+         * Calling this method has the effect of instructing the service
+         * provider to set {@code fileName} as the filename in the {@code
+         * Content-Disposition} header of the response to a request issued with
+         * a URI obtained by calling {@link
+         * BinaryDownload#getURI(BinaryDownloadOptions)}.  This value can be
+         * later retrieved by calling {@link
+         * BinaryDownloadOptions#getFileName()} on the instance returned by a
+         * call to {@link #build()}.
+         * <p>
+         * If no filename is provided, no {@code Content-Disposition} header
+         * will be specified to the service provider.
```

Comment:
I think we'll need to define the case of no filename but non-default disposition type.